src/coord/src/sink_connector.rs

```diff
@@ -118,34 +118,66 @@ async fn get_latest_ts(
     // Seek to end-1 offset
     let mut tps = TopicPartitionList::new();
     tps.add_partition(consistency_topic, partition);
-    tps.set_partition_offset(consistency_topic, partition, Offset::OffsetTail(1))?;
+    tps.set_partition_offset(consistency_topic, partition, Offset::Beginning)?;
 
-    consumer
-        .assign(&tps)
-        .with_context(|| format!("Error seeking in consistency topic {}", consistency_topic))?;
+    consumer.assign(&tps).with_context(|| {
+        format!(
+            "Error seeking in consistency topic {}:{}",
+            consistency_topic, partition
+        )
+    })?;
+
+    // We scan from the beginning and see if we can find an END record. We have
+    // to do it like this because Kafka Control Batches mess with offsets. We
+    // thereforecannot simply take the last offset from the back and expect an
+    // END message there. With a transactional producer, the OffsetTail(1) will
+    // not point to an END message but a control message. With aborted
+    // transactions, there might even be a lot of garbage at the end of the
+    // topic or in between.
+
+    let mut latest_message = None;
+    while let Some(message) = get_next_message(consumer, timeout)? {
+        latest_message = Some(message);
+    }
 
-    // Read the end-1 offset message
-    let m = match get_next_message(consumer, timeout)? {
+    let latest_message = match latest_message {
         None => {
             // fetch watermarks to distinguish between a timeout reading end-1 and an empty topic
             match consumer.fetch_watermarks(consistency_topic, 0, timeout) {
-                Ok((_, hi)) => {
+                Ok((lo, hi)) => {
                     if hi == 0 {
                         return Ok(None);
                     } else {
-                        bail!("uninitializedlkajsdlkfjs");
+                        bail!(
+                            "uninitialized consistency topic {}:{}, lo/hi: {}/{}",
+                            consistency_topic,
+                            partition,
+                            lo,
+                            hi
+                        );
                     }
                 }
                 Err(e) => {
                     bail!("Failed to fetch metadata while reading from consistency topic, will retry. Error was {}", e);
```

Comment:
this error message says it will retry but i don't think it will?

Reply:
I think it might not retry but that's how it was before. Was there maybe some retry logic in the coordinator that we don't have anymore?

Reply:
Talked to @elindsey: this is a leftover from refactoring. Will put another commit on top and then merge.